packages/vite/src/node/optimizer/esbuildDepPlugin.ts

```diff
@@ -115,7 +115,10 @@ export function esbuildDepPlugin(
         namespace: 'optional-peer-dep',
       }
     }
-    if (environment.config.consumer === 'server' && isBuiltin(resolved)) {
+    if (
+      environment.config.consumer === 'server' &&
+      isBuiltin(resolved, environment.config)
+    ) {
```

Comment:
Note: I think this could be cleaned up if we do consider this new option, the client environment's `isBuiltin` method could simply always return `false` making this code more generic (removing the need to know that this is running on the server):
```diff
-      environment.config.consumer === 'server' &&
-      isBuiltin(resolved, environment.config)
+      isBuiltin(resolved, environment.config)
```